fabric-model-events-api-v1/src/client/java/net/fabricmc/fabric/api/client/modelevents/v1/ModelPartCallbacks.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.client.modelevents.v1;
+
+import net.fabricmc.fabric.impl.client.modelevents.ModelPartCallbacksImpl;
+import net.minecraft.block.entity.BlockEntity;
+import net.minecraft.block.entity.BlockEntityType;
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.EntityType;
+
+/**
+ * Provides events for when individual model parts are rendered.
+ * <p>
+ * Example usage:
+ *
+ * <pre>
+ * {@code
+ * ModelPartCallbacks.get(PartTreePath.of("head")).register(EntityTypes.PLAYER, (player, playerRenderer, partView, matrixStack, vertexConsumerProvider, tickDelta, light) -> {
+ *      Vector3f headCenter = partView.cubes().getFirst().getCenter();
+ *      matrixStack.push();
+ *      matrixStack.translate(headCenter.x, headCenter.y, headCenter.z);
+ *      // Draw your model
+ *      // Remember to pop your stack
+ *      matrixStack.pop();
+ * });
+ * </pre>
+ * }
+ */
+public interface ModelPartCallbacks {
+    /**
+     * Gets a callbacks registry to register events upon a certain part is rendered.
```

Comment:
```suggestion
     * Gets a callbacks registry to register events triggered upon rendering a certain part.
```

Registration occurs before rendering. Also, upon is a preposition.

Reply:
Reworded and added more detail as it's "parts matching a path" rather than just "a certain part".